test/package_list_files_test.dart

```diff
@@ -344,6 +344,23 @@ void main() {
     });
   });
 
+  test('.-files are ignored', () async {
+    ensureGit();
+    final repo = d.git(appPath, [
+      d.appPubspec(),
+      d.dir('.dart_tool', [d.file('package_config.json')]),
+      d.dir('.github', [d.file('ignored.yml')]),
+      d.file('.gitignore'),
+    ]);
+
+    await repo.create();
+    createEntrypoint();
+
+    expect(entrypoint.root.listFiles(), {
+      p.join(root, 'pubspec.yaml'),
+    });
```

Comment:
Let's not add unit tests... it's better to write an integration test where we use `pub publish` and then look at the list that `pub publish` prints to stdout...

Or even better looks the actual files in the tarball published by `pub publish`.

Reply:
I will try to do it, Thank you.